### ISSUE TITLE
Solve Travis CI timeout for Clang

### ISSUE THREAD

**Issue by cbielow** [OpenMS/OpenMS#2384]
we always run into the time limit barrier.

multiple ways to resolve:
 - disable it (should be done in any case, until we have a new solution to see the green light CI tests again)
 - split the tests further (not ideal since its more build system hassle)
 - improve speed of building (compiler flags, caching of packages?)
 - use another CI

**Comment by hroest** [comment]
basically it has to do with the fact that it is a debug-build with the address sanitizer. We have both a gcc build and a clang build in release mode and they both only take 30 minutes. So I think what we could do is to move the address sanitizer to cdash and only do a nightly test on it. I made a test build with clang-debug but without the address-sanitizer and lets see how well that does:

https://travis-ci.org/OpenMS/OpenMS/builds/213507338

**Comment by hroest** [comment]
@jpfeuffer is there currently a nightly cdash that has the AdressSanitizer on?

**Comment by hroest** [comment]
yes, pretty clear that the issue was the AdressSanitizer as this branch now runs pretty fast: https://github.com/OpenMS/OpenMS/commit/c08ed80e42c8f50f8e199585564a428b42045f96 and this is also what clang more or less specifies:
```
  Typical slowdown introduced by AddressSanitizer is 2x.
```
http://clang.llvm.org/docs/AddressSanitizer.html - I guess the slowdown is not just running but also building.

**Comment by hroest** [comment]
well it is also a lot the tests:

```
Total Test time (real) = 154.76 sec
Total Test time (real) = 1373.97 sec
```

so basically 23 minutes vs 2.5 minutes to run all tests...

**Comment by hroest** [comment]
fixed through #2567
